### PR TITLE
Add tests for Book model

### PR DESCRIPTION
**Pull request by sebasdeldi**
## Summary
- Adds rspec, shoulda-matchers, factory-bot, database-cleaner and faker
- Adds unit tests for book model
- Allows travis cl to run rspec

## Trello Card
https://trello.com/c/LWzYrIPW/20-crear-tests-para-el-modelo-de-book

### CODE REVIEW

File: .travis.yml [hunk 1]
```diff
@@ -28,7 +28,7 @@ before_install:
 
 script:
   # TODO allow rspec when installed and configured
```
Comment (nicozar95) [inline .travis.yml]: Remove the TODO

File: spec/factories.rb [hunk 2]
```diff
@@ -0,0 +1,10 @@
+FactoryBot.define do
+  factory :book do
+    genre { 'Horror' }
```
Comment (nicozar95) [inline spec/factories.rb]: Use [Faker](https://github.com/stympy/faker) to create the factory so it doesn't use the same values always

File: spec/models/book_spec.rb [hunk 3]
```diff
@@ -0,0 +1,13 @@
+require 'rails_helper'
+require 'factories'
+
+RSpec.describe Book, type: :model do
+  subject(:book) { build(:book) }
```
Comment (nicozar95) [inline spec/models/book_spec.rb]: `create(:book)` I know it's not a very huge change but it's preferable to test the model once it's created in the DB

**Review by nicozar95 — CHANGES_REQUESTED** [review]

File: Gemfile [hunk 4]
```diff
@@ -42,5 +42,18 @@ group :development do
   gem 'spring-watcher-listen', '~> 2.0.0'
 end
 
+group :test do
+  # Ensures a clean state during tests
+  gem 'database_cleaner'
```
Comment (Gmarn) [inline Gemfile]: Try to lock gem version, even in test ;)

File: spec/factories.rb [hunk 5]
```diff
@@ -0,0 +1,10 @@
+FactoryBot.define do
+  factory :book do
+    genre { Faker::Book.genre }
+    author { Faker::Name.name }
+    image { Faker::Book.title }
```
Comment (Gmarn) [inline spec/factories.rb]:
```suggestion
    image { Faker::Internet.url }
```

**Review by nicozar95 — APPROVED** [review]

**Pull request merged**